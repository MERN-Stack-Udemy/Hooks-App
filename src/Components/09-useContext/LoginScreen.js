import React, { useContext } from 'react'
import { useForm } from '../../Hooks/useForm';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

  const [ {userName}, handleChange ] = useForm({});

  const {setUser} = useContext(UserContext);

  const handleSubmit = (e) =>{
    e.preventDefault();
    setUser({
      id: new Date().getTime(),
      name: userName
    })
  }

  return (
    <div>
      <h1>Login Screen</h1>
      <hr/>
      <div className="container-md">
        <form onSubmit={ handleSubmit }>
          <input
            autoComplete="off"
            className="form-control"
            name="userName"
            onChange={handleChange}
            placeholder="User name"
            type="text"
            value={userName}
          />

          <button 
            className="btn btn-outline-primary mt-2 btn-block" 
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  )
}
