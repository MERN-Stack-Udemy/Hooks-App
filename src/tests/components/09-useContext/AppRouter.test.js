import { mount } from "enzyme";
import { AppRouter } from "../../../Components/09-useContext/AppRouter";
import { UserContext } from "../../../Components/09-useContext/UserContext";

describe('test on <AppRouter/>', () => {
    
    const user = {
        id: 1,
        name : 'smog'
    } 
    const wrapper = mount(
        <UserContext.Provider value={{ user }}>
            <AppRouter/>
        </UserContext.Provider>
    )
    
    test('should match to snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    })


    
})
