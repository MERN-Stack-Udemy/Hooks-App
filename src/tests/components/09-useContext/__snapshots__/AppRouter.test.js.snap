// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <AppRouter/> should match to snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      useContext
    </a>
    <div
      className="collapse navbar-collapse"
      id="navabarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </div>
    </div>
  </nav>
  <div>
    <h1>
      Home Screen
    </h1>
    <hr />
    <pre
      className="container"
    >
      {
  "id": 1,
  "name": "smog"
}
    </pre>
  </div>
</div>
`;
